### ISSUE TITLE
Other mode keybindings used incorrectly

### ISSUE THREAD

**Issue by Cody157** [VSCodeVim/Vim#625]
### What did you do?

Included in my settings is a simple shortcut (avoiding keystrokes).

``` json
"vim.otherModesKeyBindings": [
    {
        "before": [";"],
        "after": [":"]
    }
],
```

Now, if I were to try to replace a single character with a semicolon, I would use `r;`. However, when using `r`, it treats it like you're still in normal mode and instead replaces with a colon. Same results for similar mappings/replacements. It's worth noting that being in Replacement Mode (with `R`) handles it appropriately.

It's possible the following should be a separate issue - if so I can break it into one but I think they're related.

When using commands with `t` and `f` (and related, I'm assuming), it also uses the `otherModesKeyBindings` settings. Meaning `dt;`, which should delete up to the nearest semicolon, will actually delete up to the nearest colon (or do nothing if no colon exists, as is usually the case for me).
Technically, this may not be a bug since it's not called `normalModeKeyBindings`, but it's worth mentioning for the differing behavior from standard vim. I'm not actually sure _what_ mode vim considers these commands, because none of my mappings seem to affect how they work.

The above example may be related to https://github.com/VSCodeVim/Vim/issues/604, since that's one of the common issues I have with this setting.
### Technical details:
- VSCode Version: 1.4.0
- VsCodeVim Version: 0.1.7
- OS: Ubuntu

**Comment by johnfn** [comment]
Strange, something must be wrong with the logic here:

https://github.com/VSCodeVim/Vim/blob/master/src/mode/remapper.ts#L37

This one is probably not too hard if someone wants to pick it up.

**Comment by johnfn** [comment]
Wow, your call out to issue #604 was remarkably astute. I can't imagine how long it would have taken me to find that bug otherwise.

**Comment by johnfn** [comment]
@rebornix my link is out of date now :(

**Comment by Cody157** [comment]
It's been a while since I looked at this. @rebornix, I would argue this is not a feature request, because I see no mapping in vim that makes `r;` replace the current character with `:`. Same goes for `t;` and related commands. I suspect they actually live outside the modes, or in some hidden mode since I couldn't get any map command to alter them.

In case that isn't clear - in my ~/.vimrc, I commented out all lines except `nnoremap ; :`, and added  `imap ; :`. Then after starting a new instance, `r;` replaced the current character with `;`, as I expected, not `:` as would be if it obeyed either normal or insert mode mappings. Same results for `t;`. At some point in the past I tried adding several other mappings (some I didn't even know existed), and got the same results.

Arguing bug vs feature request may be pointless, I just want to show where I'm coming from with why I expect the behavior I do.

That being said - this bug seems to have stopped happening with one of the latest releases (no idea which one because I skipped several versions somehow). However, I don't know if the change is intentional/permanent because I'm currently having other problems with my mappings (https://github.com/VSCodeVim/Vim/issues/1128) that may be related to whatever changed the behavior. I don't know the code base, so it could be entirely unrelated, just a possibility since it doesn't seem like anybody _intentionally_ fixed this based on the changelog and such.
